src/main/java/com/example/shopproject/service/UserService.java

```diff
@@ -1,2 +1,69 @@
-package com.example.shopproject.service;public class UserService {
+package com.example.shopproject.service;
+
+import com.example.shopproject.entity.Role;
+import com.example.shopproject.entity.User;
+import com.example.shopproject.repository.RoleRepository;
+import com.example.shopproject.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class UserService implements UserDetailsService {
```

Comment:
better not to implement UserDetailsService if you really dont need to, I think on this stage its better to delete it